Blog post - Langfuse alternatives (#17770)
PostHog/posthog.com: contents/blog/best-langfuse-alternatives.mdx

Reviewer: I think a lot of the alternatives are missing some details about pricing. Do they have a free tier? Are they self serve? How much do they cost? This could be big differentiator.

Reviewer: Maybe we don't even need the sentence before the tables, we just let the tables speak for themselves and fill the details in the similarities and differences below. 

See how minimal Sentry alternatives is: https://posthog.com/blog/best-sentry-alternatives

---
title: 'The best Langfuse alternatives & competitors, compared'
date: 2026-06-18
author:
    - rohit-rajpal
    - natalia-amorim
rootpage: /blog
featuredImage: >-
    https://res.cloudinary.com/dmukukwp6/image/upload/posthog.com/contents/images/blog/hog_ql.png
featuredImageType: full
category: General
tags:
    - Comparisons
seo:
  metaTitle: 'The best Langfuse alternatives & competitors, compared'
  metaDescription: 'Looking for a Langfuse alternative? Compare the best LLM observability tools, including PostHog, Helicone, LangSmith, and more.'
---

**Langfuse** is a well-known [open-source LLM observability platform](/blog/best-open-source-llm-observability-tools). It combines [tracing](/docs/ai-observability/traces), [prompt management](/docs/prompt-management/prompts), [evaluations](/docs/ai-evals/evaluations), and cost tracking to help developers understand how their LLM applications behave in production.

> But just like pineapple on pizza isn't for everyone, Langfuse isn't right for every team. Some need deeper evaluation tooling, some need a more integrated analytics experience or a better way to understand how AI performance affects the overall product experience, and some may just want a cheaper alternative.

Reviewer: ```suggestion
But just like pineapple on pizza isn't for everyone, Langfuse isn't right for every team. Some need deeper evaluation tooling, a better way to understand how AI performance affects the overall product experience, or may just want a cheaper alternative.
```

In this guide, we'll compare the best Langfuse alternatives and look at where each tool excels, where it falls short, and who it's actually built for.

## 1. PostHog

- **Founded**: 2020
- **Similar to**: Amplitude, Sentry, LogRocket
- **Typical users**: Engineers and product teams
- **Typical customers**: Mid-size B2Bs and startups

![PostHog AI Observability](https://res.cloudinary.com/dmukukwp6/image/upload/Posthog_AI_Observability_428690d037.png)

### What is PostHog?

> **[PostHog](/)** is the leading platform for self-driving products. You can use our desktop ([Code](/code)), [web](/ai), [Slack](/slack), and [MCP](/mcp) products to leverage apps like [product analytics](/product-analytics), [session replay](/session-replay), [feature flags](/feature-flags), [experiments](/experiments), [error tracking](/error-tracking), [AI observability](/ai-observability), [logs](/logs), and more.

Reviewer: ```suggestion
**[PostHog](/)** is the leading platform for self-driving products. You can use our desktop ([Code](/code)), [web](/ai), [Slack](/slack), and [MCP](/mcp) products to leverage tools like [product analytics](/product-analytics), [session replay](/session-replay), [feature flags](/feature-flags), [experiments](/experiments), [error tracking](/error-tracking), [AI observability](/ai-observability), [logs](/logs), and more.
```

The free tier includes 100K AI observability events per month, with usage-based pricing beyond that.

<details>
<summary>What sets PostHog apart from Langfuse?</summary>

PostHog is the only tool here where AI observability is one piece of a full product stack (analytics, replay, error tracking, flags, experiments, and more) so you can tie an AI change to a real product metric, not just a model metric. Set up takes minutes [with the wizard](/wizard), and agents (in [PostHog Code](/code) and beyond) can act on that context to help make your product self-driving.

PostHog ships new AI observability features fast, including prompt management (beta) for versioning and experimenting with prompts without redeploying code, plus sentiment classification and trace summarization that Langfuse has no direct equivalent for.

</details>

### Key features

- [**Generations**](docs/ai-observability/generations): Monitor model performance, token usage, costs, latency, and errors across your AI features from a single view.

> - [**Traces**](docs/ai-observability/traces): Follow complete AI workflows from start to finish to understand how requests move through prompts, tools, and model calls.

Reviewer: redundant given "start to finish"
```suggestion
- [**Traces**](docs/ai-observability/traces): Follow AI workflows from start to finish to understand how requests move through prompts, tools, and model calls.
```

- PostHog connects LLM traces to product analytics covering funnels, retention, and feature adoption. Langfuse focuses primarily on LLM observability.

</details>

<details>
<summary>Main similarities between PostHog and Langfuse</summary>

- Both capture LLM traces and monitor costs, latency, and errors out of the box.
- Both support multiple framework integrations and work with OpenTelemetry.
- Both offer a free tier and usage-based pricing.
- Both can be self-hosted and are SOC 2 compliant.

</details>

### Why do companies use PostHog?

According to [reviews on G2](https://www.g2.com/products/posthog/reviews), companies use PostHog because:

1. **It replaces multiple tools:** PostHog can replace Google Analytics (web and product analytics), Sentry (error monitoring), and standalone AI observability tools. Teams can monitor LLM performance, investigate AI interactions, and analyze product usage from the same dashboard.
2. **Pricing is transparent and scalable:** Reviewers value the usage-based pricing and the generous free tier they can keep using as they grow.

> 3. **A game-changer for AI builders:** Reviewers implementing AI features single out PostHog's LLM observability tracing and monitoring for their AI functionality without standing up a separate tool, alongside the analytics and error tracking they're already running.

Reviewer: This sentence doesn't really answer the question of "Why do companies use PostHog?" it is just a statement that they "single out." Gamer changer is a bit hyperbolic I think and this is sort of the same point as #1? It's all in one. Maybe I am misunderstanding.

Author: Yeah I hear you, there's only 1 review for AI observability in G2 (and they are the ones who call us game changer so I pulled that from it lol)

I can put back the old bullet point we have, I don't think we can honestly make a further/different point about AIO here since there's just one 

<img width="1053" height="426" alt="image" src="https://github.com/user-attachments/assets/798d5b52-af8f-4924-9035-97df79974ed2" />

> #### Bottom line
>
> **PostHog** is the strongest Langfuse alternative for developers and teams who don't want their LLM data stuck in a silo. PostHog connects them to the rest of your product so you can see why a model change moved a metric and ship the fix from the same place.

<WizardCTA />

## 2. LangSmith

- **Founded**: 2023
- **Similar to**: Langfuse, Arize Phoenix
- **Typical users**: AI engineers and ML engineers
- **Typical customers**: AI startups, enterprise AI teams, LangChain users

![LangSmith Dashboard](https://res.cloudinary.com/dmukukwp6/image/upload/Langsmith_Dashboard_09ad5e1ab0.png)

### What is LangSmith?

**LangSmith** is an observability and evaluation platform for LLM applications. It has its deepest, native integration with the LangChain and LangGraph ecosystem, but works with any stack, and provides tracing, debugging, monitoring, prompt management, and evaluation tooling.

> LangSmith automatically captures traces from LangChain and LangGraph applications, including agent runs, tool calls, prompts, and model responses. It also includes evaluation workflows, annotation queues, and datasets for testing model performance.

Reviewer: We already say evaluation twice vaguely.
```suggestion
LangSmith automatically captures traces from LangChain and LangGraph applications, including agent runs, tool calls, prompts, and model responses. It also includes annotation queues and datasets for testing model performance.
```

> #### Bottom line
>
> LangSmith is one of the strongest Langfuse competitors for teams building on LangChain and LangGraph. Native integrations capture traces automatically, so there's less setup to debug and evaluate. If you want an open-source, framework-agnostic observability platform, Langfuse is the stronger choice.

## 3. Arize Phoenix

- **Founded:** 2020 (Arize AI)
- **Similar to:** Langfuse, Braintrust
- **Typical users:** AI engineers and ML engineers
- **Typical customers:** Enterprise AI teams, ML platforms, companies building agentic applications

![Arize Phoenix Dashboard](https://res.cloudinary.com/dmukukwp6/image/upload/Arize_Phoenix_Dashboard_0c05ff7970.png)

### What is Arize Phoenix?

**Arize Phoenix** is an open-source LLM observability platform focused on tracing, evaluation, and monitoring for AI applications. It's built on OpenTelemetry and OpenInference, so one-line auto-instrumentation captures every prompt, model call, retrieval step, tool call, and agent action as a nested span, and you can attach evaluation scores directly back onto those traces.

> It s is built for teams running complex AI systems in production. It integrates with frameworks such as LangChain, LlamaIndex, LangGraph, Crew AI, DSPy, and others, and includes evaluation tooling for monitoring quality over time.

Reviewer: ```suggestion
It's built for teams running complex AI systems in production. It integrates with frameworks such as LangChain, LlamaIndex, LangGraph, Crew AI, DSPy, and others, and includes evaluation tooling for monitoring quality over time.
```

- Both support OpenTelemetry-based observability workflows.

</details>

### Why do companies use Arize Phoenix?

According to [G2 reviews](https://www.g2.com/products/arize-ai/reviews), companies use Arize Phoenix because:

- **Detailed traces and span views:** Phoenix makes it easy to inspect traces, spans, sessions, and trace trees, so teams can see exactly what happened at each step of a workflow.
- **Experiments and evaluations:** Phoenix includes evaluators, human annotations, and both online and offline evaluation workflows, plus datasets and experiments for testing prompts and models against consistent benchmarks.
- **Agent and tool-calling analysis:** Phoenix captures inputs, outputs, and the full sequence of tool calls and reasoning steps in multi-step agent workflows to review how an agent behaved and pinpoint where things went wrong.

> #### Bottom line
>
> Arize Phoenix is a strong Langfuse alternative for teams that prioritize observability, deep agent tracing, and evaluation workflows. Both tools cover tracing and evaluations well, but Phoenix goes further on OpenTelemetry-native instrumentation and understanding complex agent behavior.

> ## 4. Braintrust

Reviewer: Compared to LangSmith and Arize Phoenix, it seems to me that Braintrust is a lot better of an alternative to Langfuse, should we put it second?

## 5. Weights & Biases Weave

- **Founded:** 2024 (Weave)
- **Similar to:** LangSmith, Arize Phoenix
- **Typical users:** AI engineers, ML engineers, research teams
- **Typical customers:** AI startups, enterprise AI teams, ML organizations already using Weights & Biases

![Weights & Biases Weave Dashboard](https://res.cloudinary.com/dmukukwp6/image/upload/Weights_and_Biases_Weave_dashboard_ba22386509.png)

### What is Weights & Biases Weave?

**Weights & Biases Weave** is an observability and evaluation platform for LLM applications. It traces prompts, model calls, tool usage, and agent workflows, with built-in guardrails and monitors to catch quality and safety issues in production.

Weave instruments LLM calls with a single Python or TypeScript decorator, then streams traces, costs, and latency to the W&B platform, which CoreWeave acquired in 2025. Teams use it to score and compare prompt and model versions against evaluation datasets.

Their free tier for small teams includes 1 GB of Weave data ingestion per month. Pro is $50 per seat per month with higher limits; Enterprise is custom.

<details>

> <summary>What sets Braintrust apart?</summary>

Reviewer: ```suggestion
<summary>What sets Weave apart?</summary>
```

    rows={[
        'ai_observability.features.trace_visualization',
        'ai_observability.features.cost_tracking',
        'ai_observability.features.latency_tracking',
        'ai_observability.features.error_tracking',
        'ai_observability.features.sentiment_classification',
        'ai_observability.features.trace_summarization',
        'ai_observability.features.privacy_mode',
        'ai_observability.features.agent_tracing',
        'ai_observability.features.prompt_management',
        'ai_observability.features.prompt_evaluations',
        'ai_observability.features.evaluation_datasets',
        'ai_observability.features.human_annotation',
        'ai_observability.features.session_replay',
        'ai_observability.features.product_analytics',
    ]}
/>
</p>

<details>
<summary>Main differences between Weights & Biases Weave and Langfuse</summary>

- Weave includes built-in guardrails and production monitors for real-time safety and quality control. Langfuse handles this by integrating external guardrail libraries and tracing their results, rather than blocking outputs natively.
- Langfuse provides more mature prompt management and annotation workflows.

> - Langfuse is open source under the MIT license and free to self-host. Weave is part of the commercial W&B ecosystem and runs primarily on the hosted platform.

Reviewer: "runs primarily" makes it a bit unclear if there is a self-hosted option?

</details>

<details>
<summary>Main similarities between Weights & Biases Weave and Langfuse</summary>

- Both capture LLM traces and monitor costs, latency, and errors.
- Both support agent and multi-step workflow tracing.
- Both provide evaluations, datasets, and benchmarking workflows.

> 

Reviewer: Both are part of larger platforms (Langfuse is owned by ClickHouse)

</details>

### Why do companies use Weights & Biases Weave?

According to [user reviews](https://www.gartner.com/reviews/product/wandb-weave) and [community discussions](https://www.reddit.com/r/mlops/comments/1i764f6/any_thoughts_on_weave_from_wandb/), companies use Weights & Biases Weave because:

- **They already use the Weights & Biases platform:** Weave extends their existing training and experiment-tracking workflows into LLM tracing and evaluation.
- **It combines monitoring and evaluations:** Teams can trace applications, run evaluations, and manage datasets from one platform.
- **It supports agent traceability:** Weave provides monitoring, evaluations, datasets, and user feedback workflows for AI applications.

> #### Bottom line
>

> > Weights & Biases Weave is one of the strongest Langfuse competitors for teams already invested in the Weights & Biases ecosystem. Observability, evaluations, datasets, and experimentation fit naturally into existing W&B workflows. If you want a dedicated open-source platform focused on LLM engineering, Langfuse is the stronger choice.

Reviewer: If you're already using W&B I'd say its the strongest there 😅 
```suggestion
> Weights & Biases Weave is a strong Langfuse competitor for teams already invested in the Weights & Biases ecosystem. Observability, evaluations, datasets, and experimentation fit naturally into existing W&B workflows. If you want a dedicated open-source platform focused on LLM engineering, Langfuse is a better choice.
```

## Which Langfuse alternative should you choose?

- Want AI observability connected to product analytics, session replay, feature flags, experiments, and the rest of your product data? **PostHog** is the strongest choice.
- Already building on LangChain or LangGraph and want the tightest native integration? **LangSmith** is hard to beat.
- Need open-source observability with strong tracing, prompt management, evaluations, and self-hosting? **Langfuse** remains one of the best options.
- Want evaluation-driven development with a heavy focus on benchmarks, experiments, and model quality? **Braintrust** is worth a look.
- Need deep agent tracing, OpenTelemetry-native instrumentation, and strong evaluation tooling? **Arize Phoenix** is a strong Langfuse alternative.

> - Looking for quick implementation, proxy-based monitoring, and built-in caching? **Helicone** was a great fit, though its maintenance-mode status makes it harder to recommend for new projects.  **Portkey** is the closest actively-developed replacement.

Reviewer: ```suggestion
- Looking for quick implementation, proxy-based monitoring, and built-in caching? Other options not detailed here might be better. **Helicone** was a great fit, though its maintenance-mode status makes it harder to recommend for new projects. **Portkey** is the closest actively-developed replacement.
```

- Already invested in the Weights & Biases ecosystem? **Weave** lets you add observability and evaluations without introducing another platform.

## Is PostHog right for you?

Here's the (short) sales pitch.

We're biased, but we think PostHog is a strong Langfuse replacement if:

- You want to understand both AI performance and user behavior on the same platform.
- You want tools to help you build a better product, not only monitor LLMs (like product analytics, session replay, feature flags, experiments, surveys, and more).
- You value transparency. We're open source and open core.
- You want to try before you buy. We're self-serve with a [generous free tier](/pricing).

> It's completely free to get started – no credit card required. Our [setup wizard](/wizard) handles configuration in minutes, or you can [check out our docs](/docs) to do it yourself.

Reviewer: ```suggestion
It's completely free to get started – no credit card required. Our [setup wizard](/wizard) handles configuration in minutes, or you can [check out our docs](/docs/ai-observability) to do it yourself.
```

<WizardCTA />

## Frequently asked questions

<details>
<summary>What is Langfuse used for?</summary>

Langfuse is an open-source platform for [LLM observability](/blog/what-is-ai-observability). You use it to trace requests through your AI app, track cost and latency per call, version your prompts, and run evals on model outputs. It's built for engineers debugging and improving LLM features.

</details>

<details>
<summary>Why look for a Langfuse alternative?</summary>

Common reasons include [wanting AI observability alongside product analytics and session replay](/blog/ai-observability-for-mvps), needing tighter LangChain or LangGraph integration, prioritizing evaluations and experimentation, or building agents that require deeper tracing and debugging workflows.

</details>

<details>
<summary>What's the best Langfuse alternative overall?</summary>

> **PostHog** is the strongest choice if you want AI observability connected to product analytics and session replay. It's the only tool in this comparison that combines AI observability with product analytics, session replay, feature flags, and other product-development tools. If your priority is prompt management, evaluation workflows, and annotation queues, Langfuse remains the stronger option.

Reviewer: ```suggestion
**PostHog** is the strongest choice if you want AI observability connected to product analytics and session replay. It's the only tool in this comparison that combines AI observability with product analytics, session replay, feature flags, and other product-development tools. It also has a more generous free tier.

If your priority is prompt management, evaluation workflows, and annotation queues, Langfuse remains the stronger option.
```

</details>

<details>
<summary>Can PostHog replace Langfuse?</summary>

For some teams, yes. PostHog covers tracing, cost tracking, latency monitoring, and evaluations, with prompt management in beta. Langfuse is stronger on annotation queues, evaluation datasets, and detailed agent tracing. If your priority is connecting AI performance to user behavior, PostHog is the better fit.

</details>

<details>
<summary>Is there a free or open-source Langfuse alternative?</summary>

**PostHog** and **Arize Phoenix** are both open source, self-hostable, and free to start. PostHog's free tier includes 100K LLM observability events per month, plus product analytics, session replay, and feature flags. Arize Phoenix is free and open source with no usage limits on the self-hosted version.

For more alternatives, check out our list of [cheapest AI observability tools](/blog/cheapest-ai-observability-tools).

</details>

<details>

> <summary>Any open-source alternatives to LangSmith for tracing and debugging?</summary>

Reviewer: It's a little weird having a LangSmith one in here, maybe at the end? It seems like a typo at the moment.

**Langfuse** and **Arize Phoenix** are the strongest open-source alternatives. Langfuse covers tracing, prompt management, and evals, and is fully self-hostable. Arize Phoenix is OpenTelemetry-native and provides deep visibility into agent and multi-step workflows. **PostHog** is also open source and adds product analytics and session replay on top of tracing.

> 

Reviewer: Link? 

https://posthog.com/blog/best-open-source-llm-observability-tools

</details>

<details>
<summary>What's the best Langfuse alternative for small teams or startups?</summary>

**PostHog** is a strong fit. The free tier includes 100K LLM observability events per month, plus product analytics, session replay, feature flags, and experiments, so you get multiple tools in the same platform. If you're building on LangChain, **LangSmith** offers tight native integration with a free starter plan.

</details>

<details>
<summary>Which Langfuse alternative is best for LangChain?</summary>

**LangSmith**. It's built by the LangChain team, so tracing, debugging, and evaluating LangChain and LangGraph applications is tightly integrated from day one.

</details>

<details>
<summary>Which Langfuse alternative is best for AI evaluations?</summary>

**Braintrust** is one of the strongest options for teams focused on evaluations. It's built around evaluations, benchmarks, and experiments, so measuring model quality and comparing changes is the core workflow rather than an add-on. **LangSmith**, **Arize Phoenix**, and **Weave** also offer strong evaluation tooling.

</details>

<details>
<summary>What other AI observability tools are available?</summary>

> Beyond the tools in this guide, there are plenty of other options worth exploring. See our roundup of the [best AI observability tools](https://posthog.com/blog/best-open-source-llm-observability-tools) for a broader comparison.

Reviewer: ```suggestion
Beyond the tools in this guide, there are plenty of other options worth exploring. See our roundup of the [best AI observability tools](/blog/best-open-source-llm-observability-tools) for a broader comparison.
```

seo:
  metaTitle: 'The best Langfuse alternatives & competitors, compared'
  metaDescription: 'Looking for a Langfuse alternative? Compare the best LLM observability tools, including PostHog, Helicone, LangSmith, and more.'
---

**Langfuse** is a well-known [open-source LLM observability platform](/blog/best-open-source-llm-observability-tools). It combines [tracing](/docs/ai-observability/traces), [prompt management](/docs/prompt-management/prompts), [evaluations](/docs/ai-evals/evaluations), and cost tracking to help developers understand how their LLM applications behave in production.

But just like pineapple on pizza isn't for everyone, Langfuse isn't right for every team. Some need deeper evaluation tooling, a better way to understand how AI performance affects the overall product experience, or may just want a cheaper alternative.

In this guide, we'll compare the best Langfuse alternatives and look at where each tool excels, where it falls short, and who it's actually built for.

## 1. PostHog

- **Founded**: 2020
- **Similar to**: Amplitude, Sentry, LogRocket
- **Typical users**: Engineers and product teams
- **Typical customers**: Mid-size B2Bs and startups

> ![PostHog AI Observability](https://res.cloudinary.com/dmukukwp6/image/upload/Posthog_AI_Observability_428690d037.png)

Author: ```suggestion
![PostHog AI Observability](https://res.cloudinary.com/dmukukwp6/image/upload/w_1600,c_limit,q_auto,f_auto/Clean_Shot_2026_06_22_at_13_19_01_2x_acc4617c50.png)
```

---
title: 'The best Langfuse alternatives & competitors, compared'
date: 2026-06-18
author:
    - rohit-rajpal
    - natalia-amorim
rootpage: /blog
featuredImage: >-
    https://res.cloudinary.com/dmukukwp6/image/upload/posthog.com/contents/images/blog/hog_ql.png
featuredImageType: full
category: General
tags:
    - Comparisons
seo:
  metaTitle: 'The best Langfuse alternatives & competitors, compared'
  metaDescription: 'Looking for a Langfuse alternative? Compare the best LLM observability tools, including PostHog, Helicone, LangSmith, and more.'
---

> **Langfuse** is one of the most known [open-source LLM observability platforms](/blog/best-open-source-llm-observability-tools). It combines [tracing](/docs/ai-observability/traces), [prompt management](/docs/prompt-management/prompts), [evaluations](/docs/ai-evals/evaluations), and cost tracking to help developers understand how their LLM applications behave in production.

Reviewer: "most known" is not the right word, it's kind of awk

> But [just like pineapple on pizza isn't for everyone](https://www.linkedin.com/posts/posthog_does-pineapple-belong-on-pizza-activity-7369794498257797120-SUKL/), Langfuse isn't right for every team. As AI applications grow more complex, some teams need deeper visibility into agent behavior, stronger evaluation workflows, or a better way to understand how AI performance affects the overall product experience.

Reviewer: I don't think we need this link

Also, I would think of Langfuse as one of the deepest and strongest alternatives. Others might be cheaper or more focused on evals, but I'd expect Langfuse to be one of the best. 

```suggestion
But just like pineapple on pizza isn't for everyone, Langfuse isn't right for every team. As AI applications grow more complex, some teams need deeper visibility into agent behavior, stronger evaluation workflows, or a better way to understand how AI performance affects the overall product experience.
```

In this guide, we'll compare the best Langfuse alternatives and look at where each tool excels, where it falls short, and who it's actually built for.

## 1. PostHog

- **Founded**: 2020

> - **Similar to**: Amplitude, Sentry, LogRocket

Reviewer: These should be tools in this list

Reviewer: Still true

Author: WDYM? I'm confused, you think Amplitude, Sentry, and Logrocket should be featured in best langfuse alternatives? 

They're not direct competitors and don't show up in any other similar listicles I've read for comparison. I agree with adding all the _obvious_ competitors but none of these 3 are obvious options, so we would just be giving them a shout and helping lift them up in a space they don't compete in

Google fully doesn't recognize it either for now 

<img width="674" height="173" alt="image" src="https://github.com/user-attachments/assets/7dedf961-758f-420f-9191-a3dc7ec89ecf" />

<img width="674" height="197" alt="image" src="https://github.com/user-attachments/assets/8387ec89-4032-4f06-a348-b75d3989f45c" />

<img width="674" height="197" alt="image" src="https://github.com/user-attachments/assets/41187492-bbbd-46d7-8386-c4d769871cf7" />

Reviewer: I mean we should say PostHog is like Langfuse and Braintrust basically rather than tools not listed as part of these alternatives.

- **Typical users**: Engineers and product teams
- **Typical customers**: Mid-size B2Bs and startups

> ![PostHog AI Observability](https://res.cloudinary.com/dmukukwp6/image/upload/Posthog_AI_Observability_428690d037.png)

Reviewer: I think we have a better, newer view than this. 
```suggestion
![PostHog AI Observability](https://res.cloudinary.com/dmukukwp6/image/upload/w_1600,c_limit,q_auto,f_auto/Clean_Shot_2026_06_22_at_13_19_01_2x_acc4617c50.png)
```

Reviewer: This didn't change?

> ### What is PostHog?

Author: @ivanagas two birds one stone – instead of opening a PR just for the positioning stuff I figured I'd edit it directly here since I was already reviewing this freelance article submission

So I edited the _what is posthog_ section and the bottom of the _Why do companies use posthog_ section

lmk your thoughts

> **[PostHog](/)** is the leading platform for self-driving products. It pairs the full context of your data ([events](/product-analytics), [errors](/error-tracking), [logs](/logs), [traces](/ai-observability), [replays](/session-replay), and more) with agents that enable you to ship better products, faster. Agents can work reactively (prompted via [Slack](/slack)), proactively (via [the Inbox](/code)), or be wired into your own editor or agent via our [MCP](/mcp), so your product becomes entirely self-driving.

Author: I know leading platform may feel a tad self-patty-in-the-back BUT I think having a qualifier makes platform feel less random, and it doubles down on the fact we're leading the charge here

Alternatively could do "pioneering platform for self-driving products" but idk pioneering feels a bit overused as well

not married to either, thoughts feelings?

Reviewer: me when a product gets called pioneering

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/696b3e29-352c-4968-b302-0e2b3aabc39e" />

> **[PostHog](/)** is the leading platform for self-driving products. It pairs the full context of your data ([events](/product-analytics), [errors](/error-tracking), [logs](/logs), [traces](/ai-observability), [replays](/session-replay), and more) with agents that enable you to ship better products, faster. Agents can work reactively (prompted via [Slack](/slack)), proactively (via [the Inbox](/code)), or be wired into your own editor or agent via our [MCP](/mcp), so your product becomes entirely self-driving.

Author: ```suggestion
**[PostHog](/)** is the leading platform for self-driving products. You can use our desktop ([Code](/code)), [web](/ai), [Slack](/slack), and [MCP](/mcp) products to leverage apps like [product analytics](/product-analytics), [session replay](/session-replay), [feature flags](/feature-flags), [experiments](/experiments), [error tracking](/error-tracking), [AI observability](/ai-observability), [logs](/logs), and more.
```

alt based on slack

> You can correlate AI feature usage with retention and other product metrics. You can watch session replays of users interacting with AI features, query trace data with SQL or [through the MCP server](/docs/ai-observability/query-traces-mcp) directly from your editor, and investigate issues without jumping between tools.

Reviewer: Rather than focusing on how the AI features combine with other features of PostHog, I would expect this to be focused on PostHog's AI observability features, the ones similar to Langfuse, as the first paragraph is a general overview.

PostHog [supports popular frameworks](/docs/ai-observability/installation), including OpenAI, Anthropic, LangChain, xAI, LlamaIndex, and the Vercel AI SDK. 

The free tier includes 100K AI observability events per month, with usage-based pricing beyond that.

### Key features

> - [**Generations**](docs/ai-observability/generations): Monitor model performance, token usage, costs, latency, and errors across your AI features from a single view.
> - [**Traces**](docs/ai-observability/traces): Follow complete AI workflows from start to finish to understand how requests move through prompts, tools, and model calls.
> - [**AI evals**](docs/ai-evals/evaluations): Automatically score model outputs using LLM-as-a-judge or code-based checks to track quality and identify regressions over time.
> - [**Prompt management**](docs/prompt-management) *(beta):* Create, version, and update prompts without redeploying code. Compare versions and understand how prompt changes affect outputs.
> - [**SQL access**](docs/sql): Query AI observability data with SQL and analyze it alongside product, user, and business data.

Reviewer: ```suggestion
- [**Generations**](/docs/ai-observability/generations): Monitor model performance, token usage, costs, latency, and errors across your AI features from a single view.
- [**Traces**](/docs/ai-observability/traces): Follow complete AI workflows from start to finish to understand how requests move through prompts, tools, and model calls.
- [**AI evals**](/docs/ai-evals/evaluations): Automatically score model outputs using LLM-as-a-judge or code-based checks to track quality and identify regressions over time.
- [**Prompt management**](/docs/prompt-management) *(beta):* Create, version, and update prompts without redeploying code. Compare versions and understand how prompt changes affect outputs.
- [**SQL access**](/docs/sql): Query AI observability data with SQL and analyze it alongside product, user, and business data.
```

- [**Session replay**](/docs/ai-observability/link-session-replay): Watch recordings of users interacting with AI features and investigate issues alongside the actions that triggered them.

### How does PostHog compare to Langfuse?

**Langfuse** is built for teams who want to go deep on LLM-specific tooling only. PostHog is built for teams who want to understand how their AI feature affects real users and the broader context of its business metrics.

> Another difference is what you're paying for. Both PostHog and Langfuse use usage-based pricing, but while Langfuse focuses on LLM observability, PostHog combines AI observability with analytics, session replay, feature flags, experimentation, and other product tools in the same platform.

Reviewer: This is sort of the same point. You're not really paying for either because Langfuse doesn't have the features and PostHog has a free tier.

<details>
<summary>Main differences between PostHog and Langfuse</summary>

- Langfuse includes production-ready prompt management with versioning and deployment controls. [PostHog's prompt management](/docs/prompt-management) is currently in beta.
- PostHog adds built-in [sentiment classification](docs/ai-observability/sentiment) (currently in beta) and [trace summarization](docs/ai-observability/summarization) on top of raw trace data. Langfuse has no direct, built-in equivalent for either.
- Langfuse provides more detailed agent and multi-step tracing. PostHog supports traces and spans, but its tracing capabilities are less specialized.
- PostHog connects LLM traces to product analytics covering funnels, retention, and feature adoption. Langfuse focuses primarily on LLM observability.

</details>

<details>
<summary>Main similarities between PostHog and Langfuse</summary>

- Both capture LLM traces and monitor costs, latency, and errors out of the box.
- Both support multiple framework integrations and work with OpenTelemetry.
- Both offer a free tier and usage-based pricing.
- Both can be self-hosted and are SOC 2 compliant.

</details>

### Why do companies use PostHog?

> According to [reviews on G2](https://www.g2.com/products/posthog/reviews), companies use PostHog because:

Reviewer: I don't think we need to link to G2
```suggestion
According to reviews on G2, companies use PostHog because:
```

1. **It replaces multiple tools:** PostHog can replace Google Analytics (web and product analytics), Sentry (error monitoring), and standalone AI observability tools. Teams can monitor LLM performance, investigate AI interactions, and analyze product usage from the same dashboard.
2. **Pricing is transparent and scalable:** Reviewers value the usage-based pricing and the generous free tier they can keep using as they grow.

> 3. **They get a complete picture of users:** Funnels, session replay, A/B testing, and surveys sit alongside the analytics, so teams can see behavior end to end.

Reviewer: From a quick peek at G2, is there any reviews that mention AI observability?

PostHog can also do more than just report on your product: it can act on it. By holding the full context of what's happening, agents can use that context to suggest and ship improvements. And you can steer it from wherever you already work:

- **[In Slack](/blog)** – tag `@PostHog` in any thread to ask a data question ("why did EU signups drop last week?") or to ship a fix. 
- **[In the app (PostHog AI)](/ai)** – ask questions in plain English, build dashboards, and dig into traces or replays without writing SQL.
- **[In your own editor or agent (via MCP)](/mcp)** – wire PostHog's live product context into Claude Code, Codex, or Cowork, so your agent can pull real data and take action alongside everything else.

> - **[On the desktop (PostHog Code)](/mcp)** – run coding agents on top of your product data, with signals turned into a ranked inbox and up to nine agents working in parallel.

Reviewer: I feel like this might be a bit much? It takes away from the legitmacy of the G2 reviews. 

If I was to include this, I would include in the main section. 

Also, it should be tailored to AI observability. Use a different question from "why did EU signups drop". 

```suggestion
PostHog can also do more than just report on your product: it can act on it. By holding the full context of what's happening, agents can use that context to suggest and ship improvements. And you can steer it from wherever you already work:

- **[In Slack](/slack)** – tag `@PostHog` in any thread to ask a data question ("why did EU signups drop last week?") or to ship a fix. 
- **[In the app (PostHog AI)](/ai)** – ask questions in plain English, build dashboards, and dig into traces or replays without writing SQL.
- **[In your own editor or agent (via MCP)](/mcp)** – wire PostHog's live product context into Claude Code, Codex, or Cowork, so your agent can pull real data and take action alongside everything else.
- **[On the desktop (PostHog Code)](/code)** – run coding agents on top of your product data, with signals turned into a ranked inbox and up to nine agents working in parallel.
```

### What is LangSmith?

**LangSmith** is an observability and evaluation platform for LLM applications. It has its deepest, native integration with the LangChain and LangGraph ecosystem, but works with any stack, and provides tracing, debugging, monitoring, prompt management, and evaluation tooling.

LangSmith automatically captures traces from LangChain and LangGraph applications, including agent runs, tool calls, prompts, and model responses. It also includes evaluation workflows, annotation queues, and datasets for testing model performance.

### Key features

- **Tracing:** Follow prompts, model calls, tool invocations, and agent runs through a single trace to debug LLM applications.
- **Evaluations:** Automatically score model outputs and compare prompts, models, and application versions over time.
- **Annotation queues:** Review, label, and curate model outputs to support evaluation and quality improvement workflows.
- **Datasets:** Create and manage datasets for testing prompts, models, and application changes against consistent benchmarks.

### How does LangSmith compare to Langfuse?

> LangSmith provides the deepest native integration with LangChain and LangGraph, but it is framework-agnostic and works with any stack, including no framework at all. Langfuse is also framework-agnostic and supports a broad range of AI stacks, so the two overlap more than the framework angle suggests.

Reviewer: This is the second time we've repeated this which is a bit unnecessary

Another major difference is deployment. Langfuse is open source under the MIT license and is widely used by teams that want full control over their observability stack. LangSmith is a proprietary product and is most compelling for teams already invested in LangChain.

> 

Reviewer: Since langsmith is apparently missing cost tracking could we talk about that here?

- **It helps debug LLM workflows:** Teams use LangSmith to track model behavior, investigate issues, and understand unexpected outputs without manually piecing together logs.
- **It includes built-in evaluations:** LangSmith combines tracing with evaluation tooling, helping teams monitor output quality and maintain production-ready AI applications.

> #### Bottom line
>
> LangSmith is one of the strongest Langfuse competitors for teams building on LangChain and LangGraph. Native integrations capture traces automatically, so there's less setup to debug and evaluate. If you want an open-source, framework-agnostic observability platform, Langfuse is the stronger choice.

## 3. Arize Phoenix

- **Founded:** 2020 (Arize AI)
- **Similar to:** Langfuse, Braintrust
- **Typical users:** AI engineers and ML engineers
- **Typical customers:** Enterprise AI teams, ML platforms, companies building agentic applications

![Arize Phoenix Dashboard](https://res.cloudinary.com/dmukukwp6/image/upload/Arize_Phoenix_Dashboard_0c05ff7970.png)

> ### What is Arize Phoenix?

Reviewer: Since we talk multiple times about how Phoenix has deeper observability, I'd love to know what that means front and center of this section.

Arize Phoenix is an open-source LLM observability platform focused on tracing, evaluation, and monitoring for AI applications. It supports OpenTelemetry-based instrumentation and provides detailed visibility into prompts, model calls, retrieval steps, tool usage, and agent workflows.

Phoenix is built for teams running complex AI systems in production. It integrates with frameworks such as LangChain, LlamaIndex, Haystack, DSPy, and others, and includes evaluation tooling for monitoring quality over time.

### Key features

- **Tracing:** Follow prompts, retrieval steps, model calls, tool invocations, and agent workflows through a single trace.
- **Evaluations:** Run LLM-based and code-based evaluations to measure quality, correctness, and performance.
- **Agent observability:** Visualize multi-step agent workflows, tool calls, and reasoning chains.
- **OpenTelemetry support:** Capture observability data using open standards instead of proprietary instrumentation.

### How does Arize Phoenix compare to Langfuse?

> Arize Phoenix and Langfuse are both observability platforms for LLM applications, but they focus on different areas. Phoenix emphasizes deep tracing, agent observability, evaluations, and OpenTelemetry-based instrumentation.

Reviewer: ```suggestion
Phoenix emphasizes deep tracing, agent observability, evaluations, and OpenTelemetry-based instrumentation.
```

> Langfuse provides stronger prompt management and workflow tooling, while Phoenix goes deeper on observability, OpenTelemetry-native instrumentation, and agent analysis.

Reviewer: How does Phoenix go deeper on observability, it looks like Langfuse has all of the same features plus Trace summarization.

<p>
<ProductComparisonTable
    competitors={['arize_phoenix', 'langfuse']}
    rows={[
        'ai_observability.features.trace_visualization',
        'ai_observability.features.cost_tracking',
        'ai_observability.features.latency_tracking',
        'ai_observability.features.error_tracking',
        'ai_observability.features.sentiment_classification',
        'ai_observability.features.trace_summarization',
        'ai_observability.features.privacy_mode',
        'ai_observability.features.agent_tracing',
        'ai_observability.features.prompt_management',
        'ai_observability.features.prompt_evaluations',
        'ai_observability.features.evaluation_datasets',
        'ai_observability.features.human_annotation',
        'ai_observability.features.session_replay',
        'ai_observability.features.product_analytics',
    ]}
/>
</p>

<details>
<summary>Main differences between Arize Phoenix and Langfuse</summary>

> - Arize Phoenix is built around observability, evaluation, and monitoring for AI systems, while Langfuse focuses on LLM engineering workflows such as prompt management, datasets, annotation, and observability.

Reviewer: This makes it seem like Langfuse doesn't have observability, evaluation, and monitoring which is not true.

> - Arize Phoenix provides deeper visibility into agent workflows, tool calls, retrieval pipelines, and multi-step reasoning chains. Langfuse supports agent tracing but places more emphasis on the broader development workflow.

Reviewer: This also seems not true.

- **Typical customers:** AI startups, enterprise AI teams, companies building production LLM applications

![Braintrust Dashboard](https://res.cloudinary.com/dmukukwp6/image/upload/Braintrust_dashboard_0b54958363.png)

### What is Braintrust?

**Braintrust** is an AI evaluation and observability platform focused on helping teams measure, test, and improve LLM applications. It combines tracing, evaluations, datasets, prompt management, and experimentation in a single platform.

Braintrust is best known for its evaluation tooling. Teams can build datasets, run experiments, compare prompts and models, and track quality over time. The platform also includes observability features for monitoring AI systems in production.

### Key features

- **Evaluations:** Test prompts, models, and application changes against predefined quality criteria.
- **Datasets:** Create and manage datasets for benchmarking outputs and tracking regressions.
- **Tracing:** Inspect prompts, model calls, tool usage, and application workflows.
- **Experiments:** Compare prompts, models, and configurations against evaluation datasets.

### How does Braintrust compare to Langfuse?

> Braintrust and Langfuse both provide observability and evaluation tooling for LLM applications. The biggest difference is emphasis. Braintrust is built around evaluation-driven development, while Langfuse balances observability, prompt management, datasets, and workflow tooling in a broader LLM engineering platform.

Reviewer: ```suggestion
Braintrust is built around evaluation-driven development. Langfuse balances observability, prompt management, datasets, and workflow tooling in a broader LLM engineering platform.
```

> Another major difference is experimentation. Braintrust places experiments, benchmarks, and evaluation workflows at the center of the product. Langfuse includes evaluations, but its focus extends beyond model quality measurement.

Reviewer: I don't understand how the Langfuse sentence connects to this paragraph, seems to have nothing to do with "experimentation"

<p>
<ProductComparisonTable
    competitors={['braintrust', 'langfuse']}
    rows={[
        'ai_observability.features.trace_visualization',
        'ai_observability.features.cost_tracking',
        'ai_observability.features.latency_tracking',
        'ai_observability.features.error_tracking',
        'ai_observability.features.sentiment_classification',
        'ai_observability.features.trace_summarization',
        'ai_observability.features.privacy_mode',
        'ai_observability.features.agent_tracing',
        'ai_observability.features.prompt_management',
        'ai_observability.features.prompt_evaluations',
        'ai_observability.features.evaluation_datasets',
        'ai_observability.features.human_annotation',
        'ai_observability.features.session_replay',
        'ai_observability.features.product_analytics',
    ]}
/>
</p>

<details>
<summary>Main differences between Braintrust and Langfuse</summary>

> - Braintrust is built around evaluation-driven development, while Langfuse balances observability, prompt management, datasets, and evaluation workflows.

Reviewer: This is repeated
```suggestion
```

Weave also includes built-in guardrails and production monitors for safety and quality monitoring. Langfuse relies on external guardrail tools and uses tracing and evaluations to measure their effectiveness.

<p>
<ProductComparisonTable
    competitors={['weave', 'langfuse']}
    rows={[
        'ai_observability.features.trace_visualization',
        'ai_observability.features.cost_tracking',
        'ai_observability.features.latency_tracking',
        'ai_observability.features.error_tracking',
        'ai_observability.features.sentiment_classification',
        'ai_observability.features.trace_summarization',
        'ai_observability.features.privacy_mode',
        'ai_observability.features.agent_tracing',
        'ai_observability.features.prompt_management',
        'ai_observability.features.prompt_evaluations',
        'ai_observability.features.evaluation_datasets',
        'ai_observability.features.human_annotation',
        'ai_observability.features.session_replay',
        'ai_observability.features.product_analytics',
    ]}
/>
</p>

<details>
<summary>Main differences between Weights & Biases Weave and Langfuse</summary>

> - Weave includes built-in guardrails and production monitors for real-time safety and quality control. Langfuse handles this by integrating external guardrail libraries and tracing their results, rather than blocking outputs natively.

Reviewer: This is repeated

- Both provide evaluations, datasets, and benchmarking workflows.

</details>

### Why do companies use Weights & Biases Weave?

According to [user reviews](https://www.gartner.com/reviews/product/wandb-weave) and [community discussions](https://www.reddit.com/r/mlops/comments/1i764f6/any_thoughts_on_weave_from_wandb/), companies use Weights & Biases Weave because:

- **They already use the Weights & Biases platform:** Weave extends their existing training and experiment-tracking workflows into LLM tracing and evaluation.
- **It combines monitoring and evaluations:** Teams can trace applications, run evaluations, and manage datasets from one platform.
- **It supports agent traceability:** Weave provides monitoring, evaluations, datasets, and user feedback workflows for AI applications.

> #### Bottom line
>
> Weights & Biases Weave is one of the strongest Langfuse competitors for teams already invested in the Weights & Biases ecosystem. Observability, evaluations, datasets, and experimentation fit naturally into existing W&B workflows. If you want a dedicated open-source platform focused on LLM engineering, Langfuse is the stronger choice.

> ## Honorable mention: Helicone

Reviewer: I thought we said not to include Helicone because it is in maintenance mode.

## Which Langfuse alternative should you choose?

- Want AI observability connected to product analytics, session replay, feature flags, experiments, and the rest of your product data? **PostHog** is the strongest choice.
- Already building on LangChain or LangGraph and want the tightest native integration? **LangSmith** is hard to beat.
- Need open-source observability with strong tracing, prompt management, evaluations, and self-hosting? **Langfuse** remains one of the best options.
- Want evaluation-driven development with a heavy focus on benchmarks, experiments, and model quality? **Braintrust** is worth a look.
- Need deep agent tracing, OpenTelemetry-native instrumentation, and strong evaluation tooling? **Arize Phoenix** is a strong Langfuse alternative.
- Looking for quick implementation, proxy-based monitoring, and built-in caching? **Helicone** was a great fit, though its maintenance-mode status makes it harder to recommend for new projects.  **Portkey** is the closest actively-developed replacement.

> - Already invested in the Weights & Biases ecosystem? **Weave** lets you add observability and evaluations without introducing another platform.

Reviewer: I think these are solid, I'd like to see more of what makes each platform unique or special in the main paragraph. Right now, a lot of them read as "X is an observability and evals platform with X, Y, Z features" and maybe they focus more on some features in marketing but they have the same features. 

This doesn't really help people choose an alternative.

</details>

<details>
<summary>Which Langfuse alternative is best for LangChain?</summary>

**LangSmith**. It's built by the LangChain team, so tracing, debugging, and evaluating LangChain and LangGraph applications is tightly integrated from day one.

</details>

<details>
<summary>Which Langfuse alternative is best for AI evaluations?</summary>

**Braintrust** is one of the strongest options for teams focused on evaluations. It's built around evaluations, benchmarks, and experiments, so measuring model quality and comparing changes is the core workflow rather than an add-on. **LangSmith**, **Arize Phoenix**, and **Weave** also offer strong evaluation tooling.

</details>

<details>
<summary>What other AI observability tools are available?</summary>

Beyond the tools in this guide, there are plenty of other options worth exploring. See our roundup of the [best AI observability tools](https://posthog.com/blog/best-open-source-llm-observability-tools) for a broader comparison.

</details>

<details>
<summary>What's the difference between LLM observability and traditional observability?</summary>

> Traditional observability helps you monitor infrastructure, logs, errors, and application performance. [LLM observability](https://posthog.com/blog/what-is-ai-observability) (also known as AI observability) focuses on prompts, model outputs, token usage, evaluations, tool calls, and agent workflows. Most AI teams need both.

Reviewer: ```suggestion
Beyond the tools in this guide, there are plenty of other options worth exploring. See our roundup of the [best AI observability tools](/blog/best-open-source-llm-observability-tools) for a broader comparison.

</details>

<details>
<summary>What's the difference between LLM observability and traditional observability?</summary>

Traditional observability helps you monitor infrastructure, logs, errors, and application performance. [LLM observability](/blog/what-is-ai-observability) (also known as AI observability) focuses on prompts, model outputs, token usage, evaluations, tool calls, and agent workflows. Most AI teams need both.
```

Reviewer: hehe???
